StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionTableView.swift

```diff
@@ -258,6 +258,53 @@ final class InstitutionTableView: UIView {
             tableView.tableFooterView = nil
         }
     }
+
+    func showLoadingView(
+        _ show: Bool,
+        forInstitution institution: FinancialConnectionsInstitution
+    ) {
+        guard
+            let index = institutions.firstIndex(where: { $0.id == institution.id }),
+            let loadingCell = tableView.cellForRow(
+                at: IndexPath(row: index, section: 0)
+            ) as? InstitutionTableViewCell
+        else {
+            return
+        }
+        loadingCell.showLoadingView(show)
+    }
+
+    /// Grays out all visible rows except the one with `institution`.
+    func showOverlayView(
+        _ show: Bool,
+        exceptForInstitution institution: FinancialConnectionsInstitution? = nil
+    ) {
+        let exceptInstitutionCell: UITableViewCell? = {
+            if
+                let institution,
+                let index = institutions.firstIndex(where: { $0.id == institution.id }),
+                let cell = tableView.cellForRow(
+                    at: IndexPath(row: index, section: 0)
+                )
+            {
+                return cell
+            } else {
+                return nil
+            }
+        }()
+
+        tableView
+            .visibleCells
+            .forEach { visibleCell in
+                guard
+                    let visibleCell = visibleCell as? InstitutionTableViewCell,
+                    visibleCell !== exceptInstitutionCell
+                else {
+                    return
+                }
+                visibleCell.showOverlayView(show)
+            }
+    }
```

Comment:
just making sure: is the tableview locked and unscrollable while showing the overlays? and so no need to store the `exceptForInstitution` so newly scrolled in cells respect it?

Reply:
[yes](https://github.com/stripe/stripe-ios/pull/3262/files#diff-1ec94614b0451a0e18b1219f620c7457c7201769b4a0bf799cd2b612c1a3a163R120) although you make a great point that _if_ the code would be rearranged, maybe it "leaks" a case where user interaction is still enabled

I believe the way the code is arranged today is safer than disabling user interaction here, however. I'm particularly worried about how we rely on `sheetViewControllerWillDismiss` notification to hide the overlays - if, for unknown reason, that doesn't work, I'd rather have user tap with overlays on, than be blocked. All that being said, all of this is theoretical.  To the best of my knowledge, the interaction is blocked in all cases right now.